sway/tree/container.c

```diff
@@ -746,6 +747,29 @@ void container_floating_set_default_size(struct sway_container *con) {
 	free(box);
 }
 
+
+/**
+ * Indicate to clients in this container that they are participating in (or
+ * have just finished) an interactive resize
+ */
+void container_set_resizing(struct sway_container *con, bool resizing) {
+	if (!con) {
+		return;
+	}
+
+	if (con->view) {
+		if (con->view->type == SWAY_VIEW_XDG_SHELL &&
```

Comment:
I think we have all of the `xdg_shell` specific code in `sway/desktop/xdg_shell.c`. There should not be any shell specific code in this file. So to follow that pattern, `set_resizing` should be added to `sway_view_impl`, This block would then be moved to `sway/desktop/xdg_shell.c` and would be called from here with:

```c
if (con->view->impl->set_resizing) {
    con->view->impl->set_resizing(con->view, resizing);
}
```

*Aside: actually, it looks like `surface_is_popup` and `surface_at_view` both do have shell specific code. Both of those should probably be refactored in the same way, but that's a different issue.*

Reply:
Yeah, that sounds good. I'll rearrange it this way.

Only slightly related, but do you know if there's something to implement for `set_resizing` on other views? Like, do X clients have something similar that would work with Xwayland? I don't really know about X.

Reply:
I have no idea. For this PR, you can just implement it for xdg-shell and xwayland can be implemented later if needed.